buildcontext/git.go

```diff
@@ -150,17 +150,18 @@ func (gr *gitResolver) resolveEarthProject(ctx context.Context, gwClient gwclien
 		BuildFilePath:       localBuildFile.path,
 		BuildContextFactory: buildContextFactory,
 		GitMetadata: &gitutil.GitMetadata{
-			BaseDir:            "",
-			RelDir:             subDir,
-			RemoteURL:          gitURL,
-			Hash:               rgp.hash,
-			ShortHash:          rgp.shortHash,
-			Branch:             rgp.branches,
-			Tags:               rgp.tags,
-			CommitterTimestamp: rgp.committerTs,
-			AuthorTimestamp:    rgp.authorTs,
-			Author:             rgp.author,
-			CoAuthors:          rgp.coAuthors,
+			BaseDir:              "",
+			RelDir:               subDir,
+			RemoteURL:            gitURL,
+			Hash:                 rgp.hash,
+			ShortHash:            rgp.shortHash,
+			BranchOverrideTagArg: gr.gitBranchOverride != "",
```

Comment:
As per [the comment here](https://github.com/earthly/cloud-issues/issues/11#issuecomment-1467308267) we'd like to override the EARTHLY_TARGET_TAG and EARTHLY_TARGET_TAG_DOCKER ARGS if the action is triggered from a branch.

This felt like the cleanest way to pass it along, but open to other suggestions.